src/components/CarContainer.component.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import { CarRowInfo } from "../interfaces/Car";
+import { IMG_URL } from "../utils/index";
+import { FlexColumn, H3, ValidationText } from "./styled";
+import { CarInfo } from "./CarInfo.component";
+import { ApolloError } from "@apollo/client";
+import Loader from "./styled/Loader.component";
+
+interface Props {
+  data: {
+    cars: CarRowInfo[] | null;
+    error: ApolloError | undefined;
+    loading: boolean;
+  };
+}
+
+export const CarContainer = ({ data }: Props) => {
+  const { cars, error, loading } = data;
```

Comment:
if this is the carContainer or cars page, why are you getting `data` from props? wouldn't be better to handle the request in this component? or just to receive the data as prop once the request is successful and completed? also I wonder why you store cars in a state?

Reply:
I'm handling the request in the custom hook, this component is only in charge of mapping the cars from data to the page, and I'm using a state because I think that is the only way to rerender set of cars when another query rewrite the data, to be concise; this state setter is called inside a useEffect so it can listen to changes in data and rerender cars based on that.